src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,750 @@
+      "Number of nonzeros in Lagrangian Hessian.............:       25\n",
+      "\n",
+      "Total number of variables............................:       21\n",
+      "                     variables with only lower bounds:        0\n",
+      "                variables with lower and upper bounds:       10\n",
+      "                     variables with only upper bounds:        0\n",
+      "Total number of equality constraints.................:       21\n",
+      "Total number of inequality constraints...............:        0\n",
+      "        inequality constraints with only lower bounds:        0\n",
+      "   inequality constraints with lower and upper bounds:        0\n",
+      "        inequality constraints with only upper bounds:        0\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  0.0000000e+00 9.76e+04 0.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "   1  0.0000000e+00 9.61e+02 1.65e-03  -1.0 5.81e+03    -  9.90e-01 9.90e-01h  1\n",
+      "   2  0.0000000e+00 7.28e-12 9.99e+00  -1.0 5.72e+01    -  9.90e-01 1.00e+00h  1\n",
+      "\n",
+      "Number of Iterations....: 2\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Dual infeasibility......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Constraint violation....:   7.2759576141834259e-12    7.2759576141834259e-12\n",
+      "Complementarity.........:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Overall NLP error.......:   7.2759576141834259e-12    7.2759576141834259e-12\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 3\n",
+      "Number of objective gradient evaluations             = 3\n",
+      "Number of equality constraint evaluations            = 3\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 3\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 2\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.002\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    }
+   ],
+   "source": [
+    "def my_initialize(unit, **kwargs):\n",
+    "    # Callback for user provided initialization sequence\n",
+    "    unit.eq_flux.deactivate()\n",
+    "    unit.eq_cond.deactivate()\n",
+    "    unit.eq_flow.deactivate()\n",
+    "    assert degrees_of_freedom(unit) == 0\n",
+    "    if degrees_of_freedom(unit) == 0:\n",
+    "        res = solver.solve(m)\n",
+    "    unit.eq_flux.activate()\n",
+    "    unit.eq_cond.activate()\n",
+    "    unit.eq_flow.activate()\n",
```

Comment:
I do not understand what this initialization method is trying to do. You are deactivating all the pervaporation model constraints. It is not clear how the degrees of freedom become zero. Are you manually fixing the permeate and retentate flowrates somewhere? And, after you activate the equations back, how do you ensure that the solution it found earlier satisfies all the three constraints? 

I suspect the following. After you deactivate all the constraints, there is nothing to be solved, so the lines
```python
if degrees_of_freedom(unit) == 0:
    res = solver.solve(m)
```
will be solving an empty model. You can check this by printing the IPOPT log output and look at the number of variables and constraints. Let me know if you would like to discuss this in detail.